Tram-One/tram-one
# Fixes and Improvements for Hooks

## Summary
This PR contains a set of changes that were implemented when working on the beer-cart.io project. There were later validated using tests to confirm that they were in fact required (and useful).

## Changes

update mutation-observer.js
* copy validation done on effect cleanup function for effect updates

update observable-hook.js
* add third index in the result array, for a non-proxied value that doesn't trigger updates
* check that the value being written is defined before writing a value to the store

update tram-one.js
* expose raw.js as top-level function

udpate integration.test.js
* remove unused import
* add new test for raw method

create regression.test.js
* add regression test to verify cleanups 
* add regression test to verify that global obversables are not overwritten with undefined

Misc changes to support the above:
* create conflicting-global-counter.js
* create frozen-counter.js
* update mock-components/index.js
* create updatable-promise-page.js
* create updated-promise.js
